.github/workflows/commitlint.yml

```diff
@@ -0,0 +1,26 @@
+name: Commitlint
```

Comment:
> Unlike rook, samba-operator uses a single yaml for all is github-action jobs. This method allows creating nice dependencies between different jobs (as DAG), halting the entire workflow as soon as one of the jobs failed. Better meld this one into `main.yaml`.
@synarete , I am not fully convinced yet that it will be better to add the commitlint check to the main workflow file. @phlogistonjohn  what is your take on this? 
btw, @synarete , what does DAG mean?